subsys/net/l2/ethernet/arp.c

```diff
@@ -619,6 +620,16 @@ enum net_verdict net_arp_input(struct net_pkt *pkt)
 			}
 		}
 
+		/* Discard ARP request if Ethernet address is broadcast
+		 * and Source IP address is Multicast address.
+		 */
+		if (memcmp(&eth_hdr->dst, net_eth_broadcast_addr(),
+			   sizeof(struct net_eth_addr)) == 0 &&
+		    net_ipv4_is_addr_mcast(&arp_hdr->src_ipaddr)) {
+			NET_DBG("Drop, eth addr is bcast, src addr is mcast");
```

Comment:
Thanks for adding logging. But per @tbursztyka's proposal, it should be:

~~~
NET_DBG("DROP: eth addr is bcast, src addr is mcast");
~~~

I think it makes sense to follow that proposal.

Reply:
[00:00:03.120,000] <dbg> net_arp.net_arp_input: (0x566bb1c0): DROP: eth addr is bcast, src addr is mcast

Or

[00:00:03.120,000] <dbg> net_arp.net_arp_input: (0x566bb1c0): Drop: eth addr is bcast, src addr is mcast

Or

[00:00:03.120,000] <dbg> net_arp.net_arp_input: (0x566bb1c0): Drop, eth addr is bcast, src addr is mcast

Reply:
decide which one make sense?

Reply:
first one

Reply:
Oops, sorry, forgot to link to his post: https://github.com/zephyrproject-rtos/zephyr/pull/11662#discussion_r236543329 . There, he clearly shows `DROP: ` syntax. But can wait for extra confirmation, ping @tbursztyka .

Reply:
ok